[debian] 1.1.2 on s390x architecture, small/big indian :)

Hello

On the s390x architecture, This test failed.

```
______________________________ TestH5Ls.testHdf5 _______________________________

self = <silx.io.test.test_utils.TestH5Ls testMethod=testHdf5>

    def testHdf5(self):
        fd, self.h5_fname = tempfile.mkstemp(text=False)
        # Close and delete (we just want the name)
        os.close(fd)
        os.unlink(self.h5_fname)
        self.h5f = h5py.File(self.h5_fname, "w")
        self.h5f["/foo/bar/tmp"] = [1, 2, 3]
        self.h5f["/foo/bar/spam"] = [[1, 2], [3, 4]]
        self.h5f["/foo/data"] = [3.14]
        self.h5f.close()
    
        rep = h5ls(self.h5_fname)
        lines = rep.split("\n")
    
        self.assertIn("+foo", lines)
        self.assertIn("\t+bar", lines)
    
        match = r'\t\t<HDF5 dataset "tmp": shape \(3,\), type "<i[48]">'
>       self.assertMatchAnyStringInList(match, lines)

silx/io/test/test_utils.py:273: 
```

The result is like this

```
self = <silx.io.test.test_utils.TestH5Ls testMethod=testHdf5>
pattern = '\\t\\t<HDF5 dataset "tmp": shape \\(3,\\), type "<i[48]">'
list_of_strings = ['+foo', '\t+bar', '\t\t<HDF5 dataset "spam": shape (2, 2), type ">i8">', '\t\t<HDF5 dataset "tmp": shape (3,), type ">i8">', '\t<HDF5 dataset "data": shape (1,), type ">f8">', '']

    def assertMatchAnyStringInList(self, pattern, list_of_strings):
        for string_ in list_of_strings:
            if re.match(pattern, string_):
                return None
>       raise AssertionError("regex pattern %s does not match any" % pattern +
                             " string in list " + str(list_of_strings))
E       AssertionError: regex pattern \t\t<HDF5 dataset "tmp": shape \(3,\), type "<i[48]"> does not match any string in list ['+foo', '\t+bar', '\t\t<HDF5 dataset "spam": shape (2, 2), type ">i8">', '\t\t<HDF5 dataset "tmp": shape (3,), type ">i8">', '\t<HDF5 dataset "data": shape (1,), type ">f8">', '']
```

it seems to me that a small modification of the regex should solve this issue

- match = r'\t\t<HDF5 dataset "tmp": shape \(3,\), type "<i[48]">'
+ match = r'\t\t<HDF5 dataset "tmp": shape \(3,\), type "[<>]i[48]">'

--- a/src/silx/io/test/test_utils.py
+++ b/src/silx/io/test/test_utils.py
@@ -269,11 +269,11 @@ class TestH5Ls(unittest.TestCase):
         self.assertIn("+foo", lines)
         self.assertIn("\t+bar", lines)
 
-        match = r'\t\t<HDF5 dataset "tmp": shape \(3,\), type "<i[48]">'
+        match = r'\t\t<HDF5 dataset "tmp": shape \(3,\), type "[<>]i[48]">'
         self.assertMatchAnyStringInList(match, lines)
-        match = r'\t\t<HDF5 dataset "spam": shape \(2, 2\), type "<i[48]">'
+        match = r'\t\t<HDF5 dataset "spam": shape \(2, 2\), type "[<>]i[48]">'
         self.assertMatchAnyStringInList(match, lines)
-        match = r'\t<HDF5 dataset "data": shape \(1,\), type "<f[48]">'
+        match = r'\t<HDF5 dataset "data": shape \(1,\), type "[<>][<>]f[48]">'
         self.assertMatchAnyStringInList(match, lines)
 
         os.unlink(self.h5_fname)

--- a/src/silx/io/test/test_utils.py
+++ b/src/silx/io/test/test_utils.py
@@ -273,7 +273,7 @@ class TestH5Ls(unittest.TestCase):
         self.assertMatchAnyStringInList(match, lines)
         match = r'\t\t<HDF5 dataset "spam": shape \(2, 2\), type "[<>]i[48]">'
         self.assertMatchAnyStringInList(match, lines)
-        match = r'\t<HDF5 dataset "data": shape \(1,\), type "[<>][<>]f[48]">'
+        match = r'\t<HDF5 dataset "data": shape \(1,\), type "[<>]f[48]">'
         self.assertMatchAnyStringInList(match, lines)
 
         os.unlink(self.h5_fname)

